bin/direxec.sh

```diff
@@ -1,21 +0,0 @@
-#!/bin/bash
-# Thanks Camptocamp for the idea!
-# http://www.camptocamp.com/en/actualite/flexible-docker-entrypoints-scripts/
-set -e
-
-mode=$(basename $0 .sh)
-
-for dir in custom common; do
-    dir=/opt/odoo/$dir/$mode.d
-    if [ -d $dir ]; then
-        log INFO Executing contents of $dir
-        run-parts --exit-on-error $dir
-    else
-        log WARNING Skipping execution of $dir: not found
-    fi
-done
-
-if [ -n "$1" ]; then
-    set -x
-    exec "$@"
```

Comment:
Actually none of these do what you expect. I ran this inside a python container, see:

```
/ # python -c 'import os;os.system("sleep 5")' &
/ # ps aux
PID   USER     TIME   COMMAND
    1 root       0:00 sh
   16 root       0:00 python -c import os;os.system("sleep 5")
   17 root       0:00 sh -c sleep 5
   18 root       0:00 sleep 5
   19 root       0:00 ps aux
/ # python -c 'import subprocess;subprocess.check_output(["sleep","5"])' &
[1]-  Done                       python -c "import os;os.system(\"sleep 5\")"
/ # ps aux
PID   USER     TIME   COMMAND
    1 root       0:00 sh
   20 root       0:00 python -c import subprocess;subprocess.check_output(["sleep","5"])
   21 root       0:00 sleep 5
   22 root       0:00 ps aux
```

As you can see, `os.system` spawns the process inside a shell, and `subprocess.check_call` spawns it directly, but both spawn the subprocess. Check this instead:

```
/ # sh -c 'sleep 5' &
/ # ps aux
PID   USER     TIME   COMMAND
    1 root       0:00 sh
   11 root       0:00 sh -c sleep 5
   12 root       0:00 sleep 5
   13 root       0:00 ps aux
/ # sh -c 'exec sleep 5' &
[1]-  Done                       sh -c "sleep 5"
/ # ps aux
PID   USER     TIME   COMMAND
    1 root       0:00 sh
   14 root       0:00 sleep 5
   15 root       0:00 ps aux
```

By using `exec` in bash, you replace current process with the new one. The new one gets the parent's PID. Inside Docker, stopping a PID stops the container, that's why this little feature is important, so every signal you send with `docker kill` or with system's kill gets directly into the proper PID.

But I found [the solution](https://stackoverflow.com/a/6743663/1468388): 

```
/ # python -c 'import os; os.execvp("sleep", ["sleep", "5"])' &
/ # ps aux
PID   USER     TIME   COMMAND
    1 root       0:00 sh
   30 root       0:00 sleep 5
   31 root       0:00 ps aux
```